lib/puppet/type/service.rb

```diff
@@ -110,6 +110,12 @@ def insync?(current)
         provider.start
       end
 
+      newvalue(:absent)
+
+      validate do |val|
+        fail "Managing absent on a service is not supported" if val.equal?(:absent)
```

Comment:
When running the following command, I was expecting to get this error but instead I got:

```
❯ bx puppet resource service virtlogd.service ensure=absent
Error: The systemd provider can not handle attribute ensure
```

The reason is kind of subtle. First, when you specify `absent` on the command line or puppet manifest, it passed to the `validate` block as a `String` not a `Symbol`, so you'll need to change it to:

```
if val.equal?('absent')
```

Second, `val.equals?('absent')` returns true if and only if both `Strings` are the exact same object, which they aren't, though they have the same values. For example:

```
irb(main):002:0> "a".equal?("a")
=> false
irb(main):003:0> "a".eql?("a")
=> true
irb(main):004:0> "a" == "a"
=> true
irb(main):005:0> :a == :a
=> true
```

In order to compare if two strings have the same value, you'll want to use `val == 'absent'` (which is usually the same as `eql?`).

Puppet has a lot of string/symbol confusion especially because spec tests often use symbols when in reality, puppet passes the value from the manifest/command line as a string.